### PR TITLE
Update BuildFromSource.md

### LINKED ISSUE #31194: How do I build all managed code in the repo?
https://github.com/dotnet/aspnetcore/blob/main/docs/BuildFromSource.md talks about there being a build.cmd that it suggests is in the root of the repo, but there's no such build.cmd currently.  I do see one in various subdirectories, e.g. src\Servers.

### PR DESCRIPTION
**Pull request by davidfowl**
Fixes https://github.com/dotnet/aspnetcore/issues/31194

### CODE REVIEW

File: docs/BuildFromSource.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ code .
 
 ### Building on command-line
 
-When developing in VS Code, you'll need to use the `build.cmd` or `build.sh` scripts in order to build the project. You can learn more about the command line options available, check out [the section below](using-dotnet-on-command-line-in-this-repo).
+When developing in VS Code, you'll need to use the `eng\build.cmd` or `eng\build.sh` scripts in order to build the project. You can learn more about the command line options available, check out [the section below](using-dotnet-on-command-line-in-this-repo).
```
Comment (pranavkm) [inline docs/BuildFromSource.md:225]: I think this is talking about individual folders, so the previous statement was correct

File: docs/BuildFromSource.md [hunk 2]
```diff
@@ -233,13 +233,13 @@ To build a code change associated with a modification, run the build script in t
 On Windows, you can run the command script:
 
 ```powershell
-.\build.cmd
+.\eng\build.cmd
```
Comment (pranavkm) [inline docs/BuildFromSource.md:236]: Yeah, it's talking about individual folders and asking you to run the `build.cmd` located at their root

**Review by pranavkm — COMMENTED** [review]
I think except for the first one, you probably need to revert the rest

File: docs/BuildFromSource.md (hunk 1 above)
Comment (dougbu) [reply docs/BuildFromSource.md:225]: Suggest reordering the verbiage to introduce the "project areas" before talking about the build scripts in those areas. Basically, everything after this point should be about building for an area though we could call out using eng/build.sh and eng/build.cmd as a special case when you need to build the whole repo.

nit: never use backslashes when talking about build.sh scripts

**Pull request closed without merging**